packages/cloud-cognitive/src/components/TagSet/TagSet.js

```diff
@@ -32,88 +32,78 @@ export let TagSet = ({
 }) => {
   const [displayCount, setDisplayCount] = useState(3);
   const [displayedTags, setDisplayedTags] = useState([]);
-  const [overflowTags, setOverflowTags] = useState([]);
   const [allTags, setAllTags] = useState([]);
-  const [filteredAllTags, setFilteredAllTags] = useState([]);
-  const [tipOpen, setTipOpen] = useState(false);
+  const [hiddenSizingTags, setHiddenSizingTags] = useState([]);
   const [showAllModalOpen, setShowAllModalOpen] = useState(false);
-  const [search, setSearch] = useState('');
   const tagSet = useRef(null);
   const displayedArea = useRef(null);
   const sizingTags = useRef([]);
   const overflowTag = useRef(null);
-  const overflowTagContent = useRef(null);
+
+  const handleShowAllClick = () => {
+    setShowAllModalOpen(true);
+  };
 
   useEffect(() => {
-    setAllTags(
+    // clone children for use in modal
+    setAllTags(children.map((child) => React.cloneElement(child)));
+
+    // use children as sizing tags
+    setHiddenSizingTags(
       children && children.length > 0
         ? children.map((child, index) => {
             return (
               <div
-                key={`sizing-tag-${child.key}`}
+                key={index}
                 className={`${blockClass}--sizing-tag`}
                 ref={(el) => (sizingTags.current[index] = el)}>
-                {React.cloneElement(child)}
+                {child}
               </div>
             );
           })
         : []
     );
-    const newDisplayedTags = [];
-    const newOverflowTags = [];
+  }, [children]);
 
-    if (children && children.length > 0) {
-      children.forEach((child, index) => {
-        if (index < displayCount) {
-          newDisplayedTags.push(
-            <div
-              key={`displayed-tag-${child.key}`}
-              className={`${blockClass}--displayed-tag`}>
-              {React.cloneElement(child)}
-            </div>
-          );
-        } else {
-          if (newOverflowTags.length < 10) {
-            newOverflowTags.push(React.cloneElement(child));
-          }
-        }
-      });
-    }
+  useEffect(() => {
+    // clone children for use as visible and overflow tags
+    let newDisplayedTags = children.map((child) => React.cloneElement(child));
```

Comment:
This also needs protecting against there being no children. Perhaps something like:

```suggestion
    let newDisplayedTags =
      children && children.length > 0
        ? children.map((child) => React.cloneElement(child))
        : [];
```